### ISSUE TITLE
Add support of PHP 5.4 build-in uploadprogress.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#719]
This is the same as https://www.drupal.org/node/1561866 and similar to #472 (which in turn was the same as https://www.drupal.org/node/1559116). Both issues filed against D8, we got the nginx fix in backdrop, I think we should get this one too.

**Comment by quicksketch** [comment]
Looks neat. From the basic examples on the [php.net documentation](http://php.net/manual/en/session.configuration.php#ini.session.upload-progress.enabled) this looks very similar to the PECL and APC upload implementations, so I don't think it would be hard to support.

We don't specifically need to fix this against nginx, any PHP 5.4+ installation will have the capability. Like the other upload progress mechanisms, it doesn't work over FastCGI.

**Comment by MrHaroldA** [comment]
> Like the other upload progress mechanisms, it doesn't work over FastCGI.

I installed 1.3.0 today; and was kind of surprised to see the warning in the status report.

> Your server is capable of displaying file upload progress, but does not have the required libraries. It is recommended to install the PECL uploadprogress library (preferred) or to install APC.

I use Nginx 1.8.0 with Php 7.0.2 over FastCGI, so this won't work no matter what PECL or Php extension I install.

Chrome has native upload progress; but I don't know about other browsers. Could it be safe/wise to drop file upload progress from Backdrop?

**Comment by klonos** [comment]
Yeah, I was getting this error (not an error per se - a blue status message):

> Upload progress Not enabled
> Your server is capable of displaying file upload progress through APC, but it is not enabled. Add apc.rfc1867 = 1 to your php.ini configuration. Alternatively, it is recommended to use PECL uploadprogress, which supports more than one simultaneous upload.

...looking at my php info (this is a php 5.6.11 / apache 2.4.12 server) shows that ```/etc/php5/apache2/conf.d/uploadprogress.ini``` **is** loaded and trying to install it verifies the fact:

```
pecl install uploadprogress
pecl/uploadprogress is already installed and is the same as the released version 1.0.3.1
install failed
```

So, I enabled APC in my php.ini and restarted apache. his time the blue status message changed to a green ok message:

> Upload progress Enabled (APC RFC1867)
> Your server is capable of displaying file upload progress using APC RFC1867. Note that only one upload at a time is supported. It is recommended to use the PECL uploadprogress library if possible.

So my guess is that the feature is not detected somehow.

**Comment by MrHaroldA** [comment]
Both APC and Pecl are deprecated/obsolete, right? And upload progress does not work when uploading multiple files, or when you are using FastCGI.

I'd still say that we should drop upload progress support altogether.

**Comment by quicksketch** [comment]
> Is this still relevant with #3211 implemented?

Client-side JS upload progress is universal and no longer server-specific, so we don't have to build support for specific solutions any more. So no, I don't think we need to collect this information, as we won't use server-side solutions in the future.
